### ISSUE TITLE
SSH listen on ipv6 only

### ISSUE THREAD

**Issue by burgerkill212** [mail-in-a-box/mailinabox#1983]
Hello,

Currently i'm already running my MIAB instance on VPS. My VPS has ipv4 and ipv6 , and currently i'm setting the SSH only listen on ipv6 ip addr. On the MIAB web admin status check show that "SSH Login (ssh) is not running (port 22). Below the ufw status of my MIAB instance

`root@box:~# ufw status
Status: active

To                         Action      From
--                         ------      ----
53                         ALLOW       Anywhere                  
25/tcp                     ALLOW       Anywhere                  
587/tcp                    ALLOW       Anywhere                  
993/tcp                    ALLOW       Anywhere                  
995/tcp                    ALLOW       Anywhere                  
4190/tcp                   ALLOW       Anywhere                  
80/tcp                     ALLOW       Anywhere                  
443/tcp                    ALLOW       Anywhere                  
53 (v6)                    ALLOW       Anywhere (v6)             
25/tcp (v6)                ALLOW       Anywhere (v6)             
587/tcp (v6)               ALLOW       Anywhere (v6)             
993/tcp (v6)               ALLOW       Anywhere (v6)             
995/tcp (v6)               ALLOW       Anywhere (v6)             
4190/tcp (v6)              ALLOW       Anywhere (v6)             
80/tcp (v6)                ALLOW       Anywhere (v6)             
443/tcp (v6)               ALLOW       Anywhere (v6)             
22/tcp (v6)                LIMIT       Anywhere (v6)

**Comment by kiekerjan** [comment]
I reproduced the issue by adding the following to my sshd.config.
WARNING Take care that you do not lose access to your box this way!
`AddressFamily inet6`
WARNING 

In the file management/status_checks.py, the function check_service(i, service, env): contains the following code:
```
	if service["public"]:
		# Service should be publicly accessible.
		if try_connect(env["PUBLIC_IP"]):
			# IPv4 ok.
			if not env.get("PUBLIC_IPV6") or service.get("ipv6") is False or try_connect(env["PUBLIC_IPV6"]):
				# No IPv6, or service isn't meant to run on IPv6, or IPv6 is good.
				running = True

			# IPv4 ok but IPv6 failed. Try the PRIVATE_IPV6 address to see if the service is bound to the interface.
			elif service["port"] != 53 and try_connect(env["PRIVATE_IPV6"]):
				output.print_error("%s is running (and available over IPv4 and the local IPv6 address), but it is not publicly accessible at %s:%d." % (service['name'], env['PUBLIC_IP'], service['port']))
			else:
				output.print_error("%s is running and available over IPv4 but is not accessible over IPv6 at %s port %d." % (service['name'], env['PUBLIC_IPV6'], service['port']))

		# IPv4 failed. Try the private IP to see if the service is running but not accessible (except DNS because a different service runs on the private IP).
		elif service["port"] != 53 and try_connect("127.0.0.1"):
			output.print_error("%s is running but is not publicly accessible at %s:%d." % (service['name'], env['PUBLIC_IP'], service['port']))
		else:
			output.print_error("%s is not running (port %d)." % (service['name'], service['port']))
```

This code will not check for IPv6 connectivity if the IPv4 connectivity check fails. By adding
```
                elif try_connect(env["PUBLIC_IPV6"]):
                        output.print_warning("%s is only running on ipv6 (port %d)." % (service['name'], service['port']))
```
before the last else statement, a warning is inserted to inform that the service only runs on IPv6.
This was shortly tested to work in this specific case. However, no idea what edge case induced failures are introduced by this change. I also have no idea how to test this thoroughly.

**Comment by burgerkill212** [comment]
the addrfamily option should be any right ? the default config sshd is commented

**Comment by kiekerjan** [comment]
By default, AddressFamily is any. By stating this configuration entry, I wanted to indicate how I reproduced the issue. I did not know how UFW would impact the issue, so I explicitly made sshd not listen on ipv4 addresses. Again, changing this configuration option might lock you out of the box if you cannot access sshd through ipv6, although it seems that your UFW settings should do the same.

**Comment by burgerkill212** [comment]
yes correct , if i'm lockout i'm still able to login to vps console to flush the UFW. here's current SSHD config

`Port 22
#AddressFamily any
#ListenAddress 0.0.0.0
ListenAddress ::
`
